language-support/hs/bindings/src/DA/Ledger/LedgerService.hs

```diff
@@ -0,0 +1,44 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- Abstraction for LedgerService, which can be composed monadically.
+module DA.Ledger.LedgerService (
+    LedgerService, runLedgerService, makeLedgerService, TimeoutSeconds(..),
+    ) where
+
+import Prelude hiding (fail)
+import Control.Monad (ap,liftM)
+import Control.Monad.Fail (MonadFail,fail)
+import Control.Monad.IO.Class (MonadIO,liftIO)
+import DA.Ledger.Retry (ledgerRetry)
+import Network.GRPC.HighLevel.Generated
+
+newtype TimeoutSeconds = TimeoutSeconds { unTimeoutSeconds :: Int } deriving Num
```

Comment:
System.Time.Extra.Seconds?

Reply:
That's a double, and the GRPC library requires an Int.
But I changed the code to avoid the newtype wrapper which I dont think was adding anything.